src/main/scala/abalone/variant/Variant.scala

```diff
@@ -191,45 +296,36 @@ abstract class Variant private[variant] (
 
   /** Once a move has been decided upon from the available legal moves, the board is finalized
     */
-  @nowarn def finalizeBoard(board: Board, uci: format.Uci, captured: Option[Piece]): Board =
-    board
+  @nowarn def finalizeBoard(board: Board, uci: format.Uci, captured: Option[Piece]): Board = board
 
-  // TODO: Abalone. Add some sensible validation checks here if appropriate
   def valid(@nowarn board: Board, @nowarn strict: Boolean): Boolean = true
 
-  val roles: List[Role] = Role.all
+  def isIrreversible(move: Move): Boolean = move.capture != None
```

Comment:
are we using `capture` to define when a pieces is pushed off the board and 'scored'? if so this makes sense. Personally I would rewrite as `move.capture.nonEmpty` but it is functional

Reply:
yes we are using capture to define when a pieces is pushed off the board and 'scored'